python/tvm/driver/tvmc/autoscheduler.py

```diff
@@ -0,0 +1,212 @@
+    )
+    parser.add_argument(
+        "--model-format",
+        choices=frontends.get_frontend_names(),
+        help="specify input model format",
+    )
+
+
+def drive_autoschedule(args):
+    """Invoke auto-scheduling with command line arguments
+
+    Parameters
+    ----------
+    args: argparse.Namespace
+        Arguments from command line parser.
+    """
+
+    # extra arguments validation before importing the model, so that obvious errors
+    # are pointed in advance.
+    if args.rpc_tracker:
+        parsed_url = urlparse("//%s" % args.rpc_tracker)
+        rpc_hostname = parsed_url.hostname
+        rpc_port = parsed_url.port or 9090
+        logger.info("RPC tracker hostname: %s", rpc_hostname)
+        logger.info("RPC tracker port: %s", rpc_port)
+
+        if not args.rpc_key:
+            raise common.TVMCException(
+                "need to provide an RPC tracker key (--rpc-key) for remote tuning"
+            )
+
+    target = common.target_from_cli(args.target)
+    mod, params = frontends.load_model(args.FILE, args.model_format)
+
+    # min_repeat_ms should be:
+    # a. the value provided by the user, if any, or
+    # b. 0ms in case target is "cpu"; otherwise 1000ms
+    if args.min_repeat_ms is not None:
+        min_repeat_ms = args.min_repeat_ms
+    else:
+        min_repeat_ms = 0 if target.keys[0] == "cpu" else 1000
+        logger.debug("Default --min-repeat-ms for this target is %s", min_repeat_ms)
+
+    if args.rpc_tracker:
+
+        runner = auto_scheduler.RPCRunner(
+            key=args.rpc_key,
+            host=rpc_hostname,
+            port=rpc_port,
+            number=args.number,
+            repeat=args.repeat,
+            n_parallel=args.parallel,
+            timeout=args.timeout,
+            min_repeat_ms=min_repeat_ms,
+        )
+    else:
+        logger.info("starting localhost tuning")
+        runner = auto_scheduler.LocalRunner(
+            number=args.number,
+            repeat=args.repeat,
+            timeout=args.timeout,
+            min_repeat_ms=min_repeat_ms,
+        )
+
+    # Create the autoscheduler tuning options
+    tuning_options = auto_scheduler.TuningOptions(
+        num_measure_trials=args.trials,
+        measure_callbacks=[auto_scheduler.RecordToFile(args.output)],
+        runner=runner,
+        builder="local",
+        early_stopping=args.early_stopping,
+    )
+
+    # Specify hardware parameters
+    hardware_params = HardwareParams(
+        args.num_cores, args.vector_unit_bytes, args.cache_line_bytes, None, None, None, None, None
+    )
+
+    # Extract the tasks from the model
+    tasks, weights = get_tuning_tasks(
+        mod, params, target, target_host, args.desired_layout, hardware_params
+    )
+
+    # Schedule the tasks (i.e., produce a schedule for each task)
+    schedule_tasks(
+        tasks,
+        weights,
+        tuning_options,
+        args.tuning_records,
+    )
+
+
+def get_tuning_tasks(
```

Comment:
This function looks unnecessary, and it's a bit weird to perform layout conversion in a function called "get_tuning_tasks".

Reply:
File has been deleted